src/server/main_service.cc

```diff
@@ -590,6 +600,42 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   GenericFamily::Init(&pp_);
   server_family_.Init(acceptor, std::move(listeners), &cluster_family_);
 
+  if (string flag = GetFlag(FLAGS_replicaof); !flag.empty()) {
+    string ip, port;
+    {
+      auto pos = flag.find_last_of(':');
+      CHECK(pos != string::npos) << "Invalid format for --replicaof: missing ':'.";
+
+      string ip1 = flag.substr(0, pos);
```

Comment:
Use `string_view` here to avoid allocation for `ip1`

Reply:
I actually specifically wanted to avoid \`string_view\` as I wanted to clear unneeded resources (such as \`ip1\` which is not needed when running the \`Await\`) - there is no way of moving a \`string_view\`, as far as I'm aware.

I guess this is quite minor nitpick though.

Reply:
It is a minor nitpick, which is pretty unimportant given this happens during startup :)
I don't understand your comment though. All you do with `ip1` is `substr()`, and `Await()` will not end the scope, so what is the issue with `string_view`?

Reply:
Later the IP and port are used to construct \`CmdArgList\`, and it is not possible to construct it using a \`string_view\`, only a \`string\`. For that reason I don't believe its possible to avoid some copying.